content/rsf-spec/index.md

```diff
@@ -0,0 +1,264 @@
+say `GET /register.rsf` is that we are not providing the same information when
+querying `GET /register.json`.
+
+What is a good name for a resource that represents the whole raw database?
+
+```
+# RSF
+
+GET /{db resource}.rsf
+
+GET /{db resource}
+Accept: application/vnd.rsf
+
+
+# JSON Lines — Hypothetical
+
+GET /{db resource}.jsonl
+
+GET /{db resource}.jsonl
+Accept: application/x-ndjson
+```
+
+
+### Commands
+
+#### <a id="assert-root-hash-command">`assert-root-hash` command</a>
+
+Asserts that the provided root hash is the same as the one computed from the
+current entry log as defined in the [Digital Proofs][digital-proofs]
+specification.
+
+##### Arguments
+
+1. The `hash` of the root of the tree.
+
+For example, the empty root hash:
+
+```
+assert-root-hash	sha-256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+```
+
+#### <a id="add-item-command">`add-item` command</a>
+
+Adds a new [Item resource][item-res] to the register. There must be at least
+an [`append-entry` command](#append-entry-command) referencing the item's hash
+later on to make the RSF patch [valid](#validation-rules).
+
+##### Arguments
+
+1. The [canonical representation][canon-rep] of the item.
+
+For exeample:
+
+```
+add-item	{"country":"GB","name":"United Kingdom","official-name":"The United Kingdom of Great Britain and Northern Ireland"}
+```
+
+#### <a id="append-entry-command">`append-entry` command</a>
+
+Appends a new [Entry resource][entry-res] to the register.
+
+##### Arguments
+
+1. The `type` of the entry determines if the entry belongs to the data log
+   (`user`) or to the metadata log (`system`).
+2. The `key` of the entry. The primary key field is the field with the same
+   name as the register.
+3. The `timestamp` of the entry. This is the time at which the entry was
+   appended to the register.
+4. The `hash` of the item for which the entry was appended. This is the
+   [sha-256 hash of the item][canon-rep].
+
+For example:
+
+```
+append-entry	user	GB	2010-11-12T13:14:15Z	sha-256:08bef0039a4f0fb52f3a5ce4b97d7927bf159bc254b8881c45d95945617237f6
+```
+
+
+### <a href="validation-rules">Validation rules</a>
+
+A RSF list of commands is expected to conform to the following rules:
+
+* [Commands](#commands) are executed in order of appearance, top to bottom.
+* Entries are numbered in sequence in order of appearance starting with 1 if
+  the register is empty, otherwise incrementing on the latest entry number
+  found in the register.
+* An [`append-entry` command](#append-entry-command) must always appear after
+  the [`add-item` command](#add-item-command) that introduces the item is
+  referencing.
+* It is illegal to have orphan items. An `add-item` must have at least one
+  `append-entry` referencing to the item.
+* It is illegal to have broken references. An `append-entry` must reference an
+  item previously introduced by an `add-item` command.
```

Comment:
In the current implementation, if a corresponding `add-item` does not appear in the RSF file then it checks to see if the item already exists in the register. If it does, then the RSF file is deemed valid, otherwise it's rejected. This means that an RSF file that has an "orphan entry" could actually be valid if it is a patch on an existing register. 

The inverse is not true for `add-item`. If an `add-item` is in an RSF file, the corresponding `append-entry` must appear in the same RSF file.

Reply:
Ah! I misunderstood this behaviour. I'll amend 👍